src/ambianic/pipeline/store.py

```diff
@@ -174,16 +177,35 @@ def notify(self, save_json: dict):
         if self.notification is None:
             return
         log.debug("Sending notification(s)..")
+        directory = "{0}/ambianic-edge/premium.yaml".format(os.getcwd())
+
         # TODO extract inference data
         for inference_result in save_json['inference_result']:
-            data = {
-                'id': save_json['id'],
-                'label': inference_result['label'],
-                'confidence': inference_result['confidence'],
-                'datetime': save_json['datetime'],
-            }
-            notification = Notification(data=data, providers=self.notification_config["providers"])
-            self.notification.send(notification)
```

Comment:
@vickywane I was referring earlier to this line. We probably need to upgrade the implementation of send() so it handles failures and schedules retries. Maybe we can do that in a separate issue and PR. WDYT?

We also need to think about throttling. In case somethings goes wrong and too many notification events are somehow pushed through the notification service. We should consider this possibly as a separate issue and PR.

Reply:
Yes, these are more advanced use cases. 

They would be implemented as this feature matures.

Reply:
OK, open issues to track these problems separately.

Reply:
@vickywane how did you address this open thread?

Reply:
It is automatically closed and I do not need to manually close it as I am using the [with](https://www.geeksforgeeks.org/with-statement-in-python/) statement not `open`.

[This](https://lerner.co.il/2015/01/18/dont-use-python-close-files-answer-depends/) article contains more explanation on this.

Reply:
@vickywane `with` only handles automatic file close. It does not address the issue of retrying a message if it failed to send, nor does it address throttling (i.e. sending too many messages to the API that cause denial of service).

Reply:
Yes  it doesnt. 

Those are quite advanced use cases. We have [this](https://github.com/ambianic/ambianic-subscriptions/issues/11) open issue within the `ambianic-subscription` repository to handle such cases.

Also, implementing throttling and email retries may likely be implemented on the Cloud API where the email is sent from using our third-party email provider.

Reply:
OK, open new separate issues for each of these scenarios and link them back to this discussion.

Reply:
Okay. The respective issues have been opened and linked to the comments.